lib/bap_disasm/bap_disasm_x86_lifter.ml

```diff
@@ -0,0 +1,2319 @@
+
+let regs_x86 : var list =
+  shared_regs
+  @ List.map ~f:(fun {v32; _} -> v32) shared_multi_regs
+  (*@ List.map ~f:(fun {v64; v32} -> v32) shared_multi_regs*)
+  @ Array.to_list (Array.sub ymms ~pos:0 ~len:8)
+
+let (r_8, r_9, r_10, r_11, r_12, r_13, r_14, r_15) = match Array.to_list nums with
+  | (r_8::r_9::r_10::r_11::r_12::r_13::r_14::r_15::[]) -> (r_8, r_9, r_10, r_11, r_12, r_13, r_14, r_15)
+  | _ -> failwith "Impossible, matching against a list of known size"
+
+let regs_x86_64 : var list =
+  shared_regs
+  @ List.map ~f:(fun {v64; _} -> v64) (shared_multi_regs @ (Array.to_list nums))
+  @ Array.to_list ymms
+
+let regs_full : var list =
+  shared_regs
+  @ List.concat (List.map ~f:(fun {v64; v32} -> [v32; v64]) shared_multi_regs)
+  @ List.map ~f:(fun {v64; _} -> v64) (Array.to_list nums)
+  @ Array.to_list ymms
+
+let regs_of_mode = function
+  | X86 -> regs_x86
+  | X8664 -> regs_x86_64
+
+let o_rax = Oreg 0
+let o_rcx = Oreg 1
+let o_rdx = Oreg 2
+let o_rbx = Oreg 3
+let o_rsp = Oreg 4
+let o_rbp = Oreg 5
+let o_rsi = Oreg 6
+let o_rdi = Oreg 7
+
+let o_es = Oseg 0
+let o_cs = Oseg 1
+let o_ss = Oseg 2
+let o_ds = Oseg 3
+let o_fs = Oseg 4
+let o_gs = Oseg 5
+
+(* let esp_e = Var esp *)
+(* and ebp_e = Var ebp *)
+(* and esi_e = Var esi *)
+(* and edi_e = Var edi *)
+(* and ecx_e = Var ecx *)
+(* and eax_e = Var eax *)
+(* and edx_e = Var edx *)
+
+let mem = nmv "mem32" (Type.mem `r32 `r8) "mem64" (Type.mem `r64 `r8)
+
+(* 32-bit registers *)
+module R32 = struct
+  let eip = rip.v32
+  let eax = rax.v32
+  let ecx = rcx.v32
+  let edx = rdx.v32
+  let ebx = rbx.v32
+  let esp = rsp.v32
+  let ebp = rbp.v32
+  let esi = rsi.v32
+  let edi = rdi.v32
+  let mem = mem.v32
+end
+
+(* 64-bit registers *)
+module R64 = struct
+  let rip = rip.v64
+  let rax = rax.v64
+  let rcx = rcx.v64
+  let rdx = rdx.v64
+  let rbx = rbx.v64
+  let rsp = rsp.v64
+  let rbp = rbp.v64
+  let rsi = rsi.v64
+  let rdi = rdi.v64
+  let mem = mem.v64
+  let r8 = r_8.v64
+  let r9 = r_9.v64
+  let r10 = r_10.v64
+  let r11 = r_11.v64
+  let r12 = r_12.v64
+  let r13 = r_13.v64
+  let r14 = r_14.v64
+  let r15 = r_15.v64
+end
+
+let cf_e = Exp.var cf
+let pf_e = Exp.var pf
+let af_e = Exp.var af
+let zf_e = Exp.var zf
+let sf_e = Exp.var sf
+let of_e = Exp.var oF
+
+let df_e = Exp.var df
+
+let seg_cs = None
+let seg_ss = None
+let seg_ds = None
+let seg_es = None
+let seg_fs = Some fs_base
+let seg_gs = Some gs_base
+
+(* eflags *)
+let df_to_offset mode e =
+  match type_of_mode mode with
+  | Type.Mem _ -> failwith "type_of_mode shouldn't be returning this"
+  | Type.Imm t ->
+    let open Exp in
+    ite (e = exp_false) (it 1 t) (it (-1) t)
+
+let bap_to_rflags =
+  let undefined d = Exp.unknown (Printf.sprintf "Undefined RFLAGS bit %d" d) r1 in
+  let unmodeled s = Exp.unknown ("Unmodeled RFLAGS bit " ^ s) r1 in
+  (List.map ~f:undefined (List.range ~stride:(-1) ~stop:`inclusive 64 32))
```

Comment:
please, refactor this to a usual outfix notation, e.g., `[x;y;z]` instead of `x :: y :: z :: []`